tokio/build.rs

```diff
@@ -1,192 +1,15 @@
-                // This compiler claims to be 1.63, but there are some nightly
-                // compilers that claim to be 1.63 without supporting the
-                // feature. Explicitly probe to check if code using them
-                // compiles.
-                //
-                // The oldest nightly that supports the feature is 2022-06-16.
-                if ac.probe_expression(AS_FD_PROBE) {
-                    enable_as_fd = true;
-                }
-            }
-        }
-
-        Err(e) => {
-            // If we couldn't detect the compiler version and features, just
-            // print a warning. This isn't a fatal error: we can still build
-            // Tokio, we just can't enable cfgs automatically.
-            println!(
-                "cargo:warning=tokio: failed to detect compiler features: {}",
-                e
-            );
-        }
-    }
-
-    if !enable_const_thread_local {
-        // To disable this feature on compilers that support it, you can
-        // explicitly pass this flag with the following environment variable:
-        //
-        // RUSTFLAGS="--cfg tokio_no_const_thread_local"
-        autocfg::emit("tokio_no_const_thread_local")
-    }
-
-    if !enable_target_has_atomic {
-        // To disable this feature on compilers that support it, you can
-        // explicitly pass this flag with the following environment variable:
-        //
-        // RUSTFLAGS="--cfg tokio_no_target_has_atomic"
-        autocfg::emit("tokio_no_target_has_atomic")
-    }
-
-    if !enable_const_mutex_new {
-        // To disable this feature on compilers that support it, you can
-        // explicitly pass this flag with the following environment variable:
-        //
-        // RUSTFLAGS="--cfg tokio_no_const_mutex_new"
-        autocfg::emit("tokio_no_const_mutex_new")
-    }
-
-    if !enable_as_fd {
-        // To disable this feature on compilers that support it, you can
-        // explicitly pass this flag with the following environment variable:
-        //
-        // RUSTFLAGS="--cfg tokio_no_as_fd"
-        autocfg::emit("tokio_no_as_fd");
-    }
-
-    if target_needs_atomic_u64_fallback {
-        // To disable this feature on compilers that support it, you can
-        // explicitly pass this flag with the following environment variable:
-        //
-        // RUSTFLAGS="--cfg tokio_no_atomic_u64"
-        autocfg::emit("tokio_no_atomic_u64")
-    }
-
     let target = ::std::env::var("TARGET").unwrap_or_default();
 
-    // We emit cfgs instead of using `target_family = "wasm"` that requires Rust 1.54.
-    // Note that these cfgs are unavailable in `Cargo.toml`.
     if target.starts_with("wasm") {
-        autocfg::emit("tokio_wasm");
         if target.contains("wasi") {
-            autocfg::emit("tokio_wasi");
+            emit("tokio_wasi");
         } else {
-            autocfg::emit("tokio_wasm_not_wasi");
+            emit("tokio_wasm_not_wasi");
         }
```

Comment:
I don't know what should I replace `tokio_wasi` with.
I thought it should be replaced by `target_os = "wasi"`, but then in `lib.rs` there's a cfg `all(target_os = "wasi", not(tokio_wasi))` which is just confusing for me.

I do want to get rid of `build.rs` though since it would make compiling tokio faster.

Reply:
The meaning of `all(target_os = "wasi", not(tokio_wasi))` is "if the target_os is wasi, then tokio_wasi must also be set". It exists to detect when `tokio_wasi` is set incorrectly.

Reply:
I do believe that you can get rid of the build script.

Reply:
Thanks!

I have now removed the `build.rs` entirely.